src/ngraph_encapsulate_op.cc

```diff
@@ -587,37 +528,27 @@ class NGraphEncapsulateOp : public OpKernel {
           input_copy_events.push_back(std::move(event_copy_input_next));
 
         } catch (const std::exception& exp) {
-          OP_REQUIRES(
-              ctx, false,
-              errors::Internal(
-                  "Caught exception while transferring tensor data to nGraph: ",
-                  exp.what(), "\n"));
+          errors::Internal(
+              "Caught exception while transferring tensor data to nGraph\n");
         } catch (...) {
-          OP_REQUIRES(ctx, false,
-                      errors::Internal(
-                          "Error in transferring tensor data to nGraph\n"));
+          errors::Internal("Error in transferring tensor data to nGraph\n");
         }
       }
       input_caches[i] = std::make_pair(current_src_ptr, current_ng_tensor);
       ng_inputs.push_back(current_ng_tensor);
     }  // for (int i = 0; i < input_shapes.size(); i++)
+    return Status::OK();
+  }
 
-    NGRAPH_VLOG(4) << "NGraphEncapsulateOp::Compute allocated argument tensors "
-                      "for cluster "
-                   << m_ngraph_cluster;
-
-    // Now write the events back
-    for (auto& next : input_copy_events) {
-      ngraph::Event::write_trace(*next.get());
-    }
-    event_alloc_input.Stop();
-
-    // Allocate tensors for the output results.
-    ngraph::Event event_alloc_output("Output: maybe create", name(), "");
-    vector<shared_ptr<ng::runtime::Tensor>> ng_outputs;
-    int ng_output_tensor_size_in_bytes = 0;
-    std::vector<std::pair<void*, std::shared_ptr<ng::runtime::Tensor>>>&
-        output_caches = m_ng_exec_output_cache_map[ng_exec];
+  Status AllocateTensorOutput(
+      OpKernelContext* ctx,
+      std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
+      std::vector<TensorShape>& input_shapes, ng::runtime::Backend* op_backend,
+      vector<shared_ptr<ng::runtime::Tensor>>& ng_inputs,
```

Comment:
Right now, `AllocateTensorOutput` accepts `ng_inputs` and `ng_outputs`. More refactor needed. `ng_inputs` is being populated in the `#if defined(NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS)` section. This should be moved to `AllocateTensorInput` in that case